protostar/testing/environments/common_test_cheatcode_factory.py

```diff
@@ -25,16 +26,17 @@
 
 
 class CommonTestCheatcodeFactory(CheatcodeFactory):
-    def __init__(self, state: TestExecutionState):
+    def __init__(self, state: TestExecutionState, project_compiler: ProjectCompiler):
```

Comment:
I think `project_compiler` could easily land in `TestExecutionState` and thus you could save a lot of passing. (Compiler by intuition **can** be a stateful object, for example if it has incremental cache).

The structure that was until now was totally fine IMHO.

Reply:
done